### PR TITLE
test: add additional test for MarshalUnmarshalBitList and refactor name comment

### LINKED ISSUE #1481: Fuzz tests verifying SSZ Marshal/Unmarshal are reversible
For example, generate random bit lists, perform `MarshalBitList`, check `UnmarshalBitList` returns the original.

### PR DESCRIPTION
**Pull request by hoank101**
Closes #1481

## Summary by CodeRabbit

- **New Features**
  - Enhanced testing for bit list serialization and deserialization.
  
- **Refactor**
  - Renamed several functions for better clarity and consistency, including `Merkleize` to `MerkleizeBasic` and `ComputeRandaoSigningRootUInt64` to `ComputeSigningRootUInt64`.

- **Style**
  - Improved comments for better code readability and understanding.

- **Documentation**
  - Updated comments to explain control flow logic in middleware, preventing potential issues.

### CODE REVIEW

**Review by itsdevbear — APPROVED** [review]

File: mod/primitives/pkg/ssz/serialization_test.go [hunk 1]
```diff
@@ -483,3 +483,93 @@ func TestUnmarshalBitList(t *testing.T) {
 		})
 	}
 }
+
+func FuzzMarshalUnmarshalBitList(f *testing.F) {
+	f.Fuzz(func(t *testing.T, data []byte) {
+		// Convert bytes to a bit list (bool slice)
+		bitList := make([]bool, len(data)*8)
```
Comment (rustybera) [inline mod/primitives/pkg/ssz/serialization_test.go:490]: Nice work. But this only produces bit lists whose length is a multiple of 8.

File: mod/primitives/pkg/ssz/serialization_test.go [hunk 2]
```diff
@@ -483,3 +483,93 @@ func TestUnmarshalBitList(t *testing.T) {
...
+			name: "all false",
+			input: []bool{false, false, false, false, false, false, false,
+				false, false, false},
+			expOutput: []byte{0b00000001},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			marshaled := ssz.MarshalBitList(tc.input)
```
Comment (rustybera) [inline mod/primitives/pkg/ssz/serialization_test.go:570]:
```go
			require.Equal(t, tc.expOutput, marshaled)
```
Add this line then some cases fail.

File: mod/primitives/pkg/ssz/serialization_test.go [hunk 3]
```diff
@@ -483,3 +483,93 @@ func TestUnmarshalBitList(t *testing.T) {
...
+			name: "fifteen elements input",
+			input: []bool{true, false, true, false, true, false, true, false, true,
+				true, true, true, true, true, true},
+			expOutput: []byte{0b01010101, 0b11111111},
+		},
+		{
+			name: "alternating pattern",
+			input: []bool{true, false, true, false, true, false, true, false,
+				true, false},
+			expOutput: []byte{0b01010101, 0b00000011},
```
Comment (rustybera) [inline mod/primitives/pkg/ssz/serialization_test.go:552]: This is not the correct result.

**Comment by hoank101** [comment]
thank @rustybera for feedback, i have PR for improve it https://github.com/berachain/beacon-kit/pull/1491

**Pull request merged**